meta-facebook/op2-op/src/platform/plat_power_seq.c

```diff
@@ -0,0 +1,783 @@
+
+LOG_MODULE_REGISTER(power_sequence);
+
+K_THREAD_STACK_EXTERN(e1s_power_thread);
+K_THREAD_STACK_ARRAY_DEFINE(e1s_power_threads, MAX_E1S_IDX, POWER_SEQ_CTRL_STACK_SIZE);
+struct k_thread e1s_power_thread_handler[MAX_E1S_IDX];
+k_tid_t e1s_power_tid[MAX_E1S_IDX];
+
+e1s_power_control_gpio opa_e1s_power_control_gpio[] = {
+	[0] = { .present = OPA_E1S_0_PRSNT_N,
+		.p12v_efuse_enable = OPA_E1S_0_12V_POWER_EN,
+		.p12v_efuse_power_good = OPA_PWRGD_P12V_E1S_0_R,
+		.p3v3_efuse_enable = OPA_E1S_0_3V3_POWER_EN,
+		.p3v3_efuse_power_good = OPA_PWRGD_P3V3_E1S_0_R,
+		.clkbuf_oe_en = OPA_CLKBUF_E1S_0_OE_N,
+		.pcie_reset = OPA_PERST_E1S_0_N },
+	[1] = { .present = OPA_E1S_1_PRSNT_N,
+		.p12v_efuse_enable = OPA_E1S_1_12V_POWER_EN,
+		.p12v_efuse_power_good = OPA_PWRGD_P12V_E1S_1_R,
+		.p3v3_efuse_enable = OPA_E1S_1_3V3_POWER_EN,
+		.p3v3_efuse_power_good = OPA_PWRGD_P3V3_E1S_1_R,
+		.clkbuf_oe_en = OPA_CLKBUF_E1S_1_OE_N,
+		.pcie_reset = OPA_PERST_E1S_1_N },
+	[2] = { .present = OPA_E1S_2_PRSNT_N,
+		.p12v_efuse_enable = OPA_E1S_2_12V_POWER_EN,
+		.p12v_efuse_power_good = OPA_PWRGD_P12V_E1S_2_R,
+		.p3v3_efuse_enable = OPA_E1S_2_3V3_POWER_EN,
+		.p3v3_efuse_power_good = OPA_PWRGD_P3V3_E1S_2_R,
+		.clkbuf_oe_en = OPA_CLKBUF_E1S_2_OE_N,
+		.pcie_reset = OPA_PERST_E1S_2_N },
+};
+
+e1s_power_control_gpio opb_e1s_power_control_gpio[] = {
+	[0] = { .present = OPB_E1S_0_PRSNT_N,
+		.p12v_efuse_enable = OPB_P12V_E1S_0_EN_R,
+		.p12v_efuse_power_good = OPB_PWRGD_P12V_E1S_0_R,
+		.p3v3_efuse_enable = OPB_P3V3_E1S_0_EN_R,
+		.p3v3_efuse_power_good = OPB_PWRGD_P3V3_E1S_0_R,
+		.clkbuf_oe_en = OPB_CLKBUF_E1S_0_OE_N,
+		.pcie_reset = OPB_RST_E1S_0_PERST },
+	[1] = { .present = OPB_E1S_1_PRSNT_N,
+		.p12v_efuse_enable = OPB_P12V_E1S_1_EN_R,
+		.p12v_efuse_power_good = OPB_PWRGD_P12V_E1S_1_R,
+		.p3v3_efuse_enable = OPB_P3V3_E1S_1_EN_R,
+		.p3v3_efuse_power_good = OPB_PWRGD_P3V3_E1S_1_R,
+		.clkbuf_oe_en = OPB_CLKBUF_E1S_1_OE_N,
+		.pcie_reset = OPB_RST_E1S_1_PERST },
+	[2] = { .present = OPB_E1S_2_PRSNT_N,
+		.p12v_efuse_enable = OPB_P12V_E1S_2_EN_R,
+		.p12v_efuse_power_good = OPB_PWRGD_P12V_E1S_2_R,
+		.p3v3_efuse_enable = OPB_P3V3_E1S_2_EN_R,
+		.p3v3_efuse_power_good = OPB_PWRGD_P3V3_E1S_2_R,
+		.clkbuf_oe_en = OPB_CLKBUF_E1S_2_OE_N,
+		.pcie_reset = OPB_RST_E1S_2_PERST },
+	[3] = { .present = OPB_E1S_3_PRSNT_N,
+		.p12v_efuse_enable = OPB_P12V_E1S_3_EN_R,
+		.p12v_efuse_power_good = OPB_PWRGD_P12V_E1S_3_R,
+		.p3v3_efuse_enable = OPB_P3V3_E1S_3_EN_R,
+		.p3v3_efuse_power_good = OPB_PWRGD_P3V3_E1S_3_R,
+		.clkbuf_oe_en = OPB_CLKBUF_E1S_3_OE_N,
+		.pcie_reset = OPB_RST_E1S_3_PERST },
+	[4] = { .present = OPB_E1S_4_PRSNT_N,
+		.p12v_efuse_enable = OPB_P12V_E1S_4_EN_R,
+		.p12v_efuse_power_good = OPB_PWRGD_P12V_E1S_4_R,
+		.p3v3_efuse_enable = OPB_P3V3_E1S_4_EN_R,
+		.p3v3_efuse_power_good = OPB_PWRGD_P3V3_E1S_4_R,
+		.clkbuf_oe_en = OPB_CLKBUF_E1S_4_OE_N,
+		.pcie_reset = OPB_RST_E1S_4_PERST },
+};
+
+uint8_t get_e1s_present(uint8_t device_index)

```

Comment:
Presence should be a bool and not a uint8_t as the drive is either present or not.

Reply:
Modify as recommendation.